### PR TITLE
Typescript: Hidden component

### PR DESCRIPTION
**Pull request by ewgenius**
Part of #219

### CODE REVIEW

File: packages/reakit/src/Hidden/HiddenContainer.tsx [hunk 1]
```diff
@@ -1,23 +1,30 @@
...
+export const toggle = () => (state: ContainerState) => ({
+  visible: !state.visible
+});
 
 export const show = () => () => ({ visible: true });
 
 export const hide = () => () => ({ visible: false });
 
-const HiddenContainer = props => (
+const HiddenContainer = (props: ContainerProps<ContainerState>) => (
```
Comment (diegohaz) [inline packages/reakit/src/Hidden/HiddenContainer.tsx]: Can we try to make it as stated in https://github.com/diegohaz/constate#typescript

Here's an example: https://github.com/reakit/reakit/blob/62b108cd9f9656512ac1bda3d166fa33196de955/packages/website/src/containers/ThemeContainer.tsx#L1-L31

File: packages/reakit/src/Hidden/index.ts [hunk 2]
```diff
@@ -4,9 +4,13 @@ import HiddenHide from "./HiddenHide";
...
 import HiddenToggle from "./HiddenToggle";
 
+// @ts-ignore
 Hidden.Container = HiddenContainer;
+// @ts-ignore
 Hidden.Hide = HiddenHide;
+// @ts-ignore
 Hidden.Show = HiddenShow;
+// @ts-ignore
 Hidden.Toggle = HiddenToggle;
```
Comment (diegohaz) [inline packages/reakit/src/Hidden/index.ts]: What do you think about this?
https://github.com/reakit/reakit/blob/62b108cd9f9656512ac1bda3d166fa33196de955/packages/reakit/src/Card/index.ts#L1-L12

File: packages/reakit/src/Hidden/HiddenContainer.tsx (hunk 1 above)
Comment (ewgenius) [reply packages/reakit/src/Hidden/HiddenContainer.tsx]: Nice 👍 
I will use this, thanks for note

File: packages/reakit/src/Hidden/index.ts (hunk 2 above)
Comment (ewgenius) [reply packages/reakit/src/Hidden/index.ts]: Interesting, definitely better then `@ts-ignore`

**Comment by diegohaz** [comment]
Hey, @ewgenius. Thank you so much for this.

I went ahead and made a few changes.

`<Hidden absolute />` wasn't working, since `absolute` is a prop from `Base` (now `Box` #225). So, it seems that we need to export the props interface in order to create derived components:
```ts
// Box/Box.ts
export interface BoxProps {
  absolute?: boolean;
}
```
```ts
// Box/index.ts
export * from "./Box";
```
```ts
// Hidden/Hidden.ts
import { BoxProps } from "../Box";

export interface HiddenProps extends BoxProps {
  visible?: boolean;
}
```

Then it has all the props from `Hidden` and `Box`.

I did this for other components too, and updated the main `index.ts` to export every member besides `default`:
https://github.com/reakit/reakit/blob/661c5d094f7bdd85bb8403fcaf41f94ff3ec375a/packages/reakit/src/index.ts#L8-L21

**Comment by ewgenius** [comment]
@diegohaz thanks for the feedback and taking care of it 👍

**Pull request merged**